src/modules/UI/scenes/CreateWallet/CreateWalletSelectCrypto.ui.js

```diff
@@ -0,0 +1,186 @@
+import SearchResults from '../../components/SearchResults'
+
+import styles, {styles as stylesRaw} from './style.js'
+import s from '../../../../locales/strings.js'
+import PLATFORM from '../../../../theme/variables/platform'
+import Gradient from '../../components/Gradient/Gradient.ui'
+import * as UTILS from '../../../utils'
+import type {GuiWalletType, FlatListItem, DeviceDimensions} from '../../../../types'
+
+const WALLET_TYPE_PICKER_PLACEHOLDER = s.strings.create_wallet_choose_crypto
+
+const BACK_TEXT = s.strings.title_back
+const INVALID_INPUT_TITLE = s.strings.create_wallet_invalid_input
+const INVALID_DATA_TEXT = s.strings.create_wallet_select_valid_crypto
+const NEXT_TEXT = s.strings.string_next_capitalized
+
+export type Props = {
+  walletName: string,
+  dimensions: DeviceDimensions,
+  supportedWalletTypes: Array<GuiWalletType>
+}
+
+export type State = {
+  selectedWalletType: string,
+  searchTerm: string
+}
+
+export class CreateWalletSelectCryptoComponent extends Component<Props, State> {
+  constructor (props: Props & State) {
+    super(props)
+    this.state = {
+      selectedWalletType: '',
+      searchTerm: ''
+    }
+  }
+
+  isValidWalletType = (): boolean => {
+    const {selectedWalletType} = this.state
+    const {supportedWalletTypes} = this.props
+    const walletTypeValue = supportedWalletTypes
+      .findIndex((walletType) => walletType.value === selectedWalletType)
+
+    const isValid: boolean = walletTypeValue >= 0
+    return isValid
+  }
+
+  getWalletType = (walletTypeValue: string): GuiWalletType => {
+    const {supportedWalletTypes} = this.props
+    const foundValueIndex = supportedWalletTypes.findIndex((walletType) => walletType.value === walletTypeValue)
+    const foundValue = supportedWalletTypes[foundValueIndex]
+
+    return foundValue
+  }
+
+  onNext = (): void => {
+    if (this.isValidWalletType()) {
+      Actions.createWalletSelectFiat({
+        walletName: this.props.walletName,
+        selectedWalletType: this.getWalletType(this.state.selectedWalletType)
+      })
+    } else {
+      Alert.alert(INVALID_INPUT_TITLE, INVALID_DATA_TEXT)
+    }
+  }
+
+  onBack = (): void => {
+    Keyboard.dismiss()
+    Actions.pop() // redirect to the list of wallets
+  }
+
+  handleSearchTermChange = (searchTerm: string): void => {
+    this.setState({
+      searchTerm
+    })
+  }
+
+  handleSelectWalletType = (item: GuiWalletType): void => {
+    const selectedWalletType = this.props.supportedWalletTypes.find((type) => type.value === item.value)
+    if (selectedWalletType) {
+      this.setState({
+        selectedWalletType: selectedWalletType.value,
+        searchTerm: selectedWalletType.label
+      })
+    }
+  }
+
+  handleOnFocus = () => {
+    UTILS.noOp()
+  }
+
+  handleOnBlur = () => {
+    UTILS.noOp()
+  }
+
+  render () {
+    const filteredArray = this.props.supportedWalletTypes.filter((entry) => {
+      return (entry.label.indexOf(this.state.searchTerm) >= 0)
```

Comment:
You should make this a case insensitive search. otherwise a user typing in "Usd" won't match with your list which has "USD".